### PR TITLE
[DataPipe] Fixing IterDataPipe docstrings

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72476
* __->__ #72475

To render the change in documentation, please pull down this PR and build the doc in `TorchData`.

Differential Revision: [D34078064](https://our.internmc.facebook.com/intern/diff/D34078064)

### CODE REVIEW

File: torch/utils/data/datapipes/iter/combining.py [hunk 1]
```diff
@@ -73,11 +72,10 @@ def __new__(cls, datapipe: IterDataPipe, num_instances: int, buffer_size: int =
 
 
 class _ForkerIterDataPipe(IterDataPipe):
```
Comment (ejguan) [inline torch/utils/data/datapipes/iter/combining.py:74]: Just want to confirm this is a private class and not shown in our doc even though we have doc strings.

File: torch/utils/data/datapipes/iter/fileopener.py [hunk 2]
```diff
@@ -6,17 +6,15 @@
...
-    Iterable Datapipe to load file streams from given pathnames,
-    yield pathname and file stream in a tuple.
+    r"""
+    Given pathnames, open files and yield pathname and file stream in a tuple.
 
     Args:
         datapipe: Iterable datapipe that provides pathnames
         mode: An optional string that specifies the mode in which
-            the file is opened by `open()`. It defaults to 'b' which
+            the file is opened by ``open()``. It defaults to ``b`` which
```
Comment (ejguan) [inline torch/utils/data/datapipes/iter/fileopener.py:15]: b and t her should be string. (Maybe: ``'b'`` and ``'t'``)

**Review by ejguan — APPROVED** [review]
Thank you so much. LGTM

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34078064).

**Pull request closed without merging**